### ISSUE TITLE
python 3.4 and below error

### ISSUE THREAD

**Issue by buuth** [rj79/pynetstring#12]
Hi,
Your requirement is Python3, but below code does not work with Python below 3.5.
Kindly fix it.
Thanks
Buu
`def _encode(data):
    if isinstance(data, str):
        data = bytes(data, 'utf-8')
    return b'%d:%s,' % (len(data), data)`

**Comment by rj79** [comment]
Thanks for bringing this to attention. I plan to fix this in the coming day.

**Comment by rj79** [comment]
Hi @buuth,

I have a small uncomitted patch that makes it work for Python 3.4. Additional changes would be needed to make it work for Python 3.3 and earlier. At least the state enums would also have to be converted into plain integers, since Enum support was introduced in Python 3.4. While this is not a big effort, even Python version 3.4 recently got end-of-life status and is no longer receiving updates.
See: https://devguide.python.org/devcycle/#end-of-life-branches

What Python version does your project use?

Thanks!

**Comment by buuth** [comment]
Thank you.  
I'm using 3.4. 
Another problem: if keep feeding data (correct data) for a while, it will eventually throw error
ERROR - Missing trailing comma.
It does not happen immediately, but just after few hours. It does not happen in 3.5 Python (16.04 Ubuntu). I have 2 server running, one in 14.04 and other one is 16.04 and problem happens only with Python 3.4 version. 
Thanks

**Comment by rj79** [comment]
Ok. I will commit the fix for Python 3.4, and update the requirements that only Python >= 3.4 is supported.
Regarding the other problem, please open a separate issue so we keep clear separate track. Also supply some minimal input data that can reproduce the problem reliably.

**Comment by buuth** [comment]
Unfortunately I cannot re-produce it locally. The issue happens in production, so I had to rollback to my own text-based parser (my data is text based). It might be difficult to re-produce, as problem emerges only after few hours. My best guess is pynetstring was feeded with some distorted data, then it's unable to recover and throw error to all valid net strings after that.
Thanks

**Comment by rj79** [comment]
Which method of decoding do you use? Is it decode() or feed()? It can happen that the data that arrives over the network breaks up a netstring in the middle. decode() can not handle that, but feed() can.

**Comment by buuth** [comment]
I use feed, definitely it has some problem with recovery after corrupted data. It just cannot recover.
